core/src/main/java/com/exadel/aem/toolkit/core/injectors/EToolboxListInjector.java

```diff
@@ -0,0 +1,144 @@
+     * @param name             Name of the Java class member to inject the value into
+     * @param type             Type of receiving Java class member
+     * @param element          {@link AnnotatedElement} instance that facades the Java class member allowing to retrieve
+     *                         annotation objects
+     * @param callbackRegistry {@link DisposalCallbackRegistry} object
+     * @return The value to inject, or null in case injection is not possible
+     * @see Injector
+     * @see ListHelper
+     */
+    @Override
+    public Object getValue(
+        @Nonnull Object adaptable,
+        String name,
+        @Nonnull Type type,
+        @Nonnull AnnotatedElement element,
+        @Nonnull DisposalCallbackRegistry callbackRegistry) {
+
+        EToolboxList annotation = element.getDeclaredAnnotation(EToolboxList.class);
+        if (annotation == null) {
+            return null;
+        }
+
+        ResourceResolver resourceResolver = InjectorUtils.getResourceResolver(adaptable);
+        if (resourceResolver == null) {
+            return null;
+        }
+
+        if (annotation.value().isEmpty()) {
+            LOG.debug(InjectorConstants.EXCEPTION_EMPTY_ANNOTATION_VALUE);
+            return null;
+        }
+
+        if (!annotation.keyProperty().isEmpty()) {
+            return InjectorUtils.isValidMapKeyType(type, String.class)
+                ? ListHelper.getMap(resourceResolver, annotation.value(), annotation.keyProperty(), getTypeArgument(type, 1))
+                : null;
+
+        } else if (InjectorUtils.isValidCollectionRawType(type, Collection.class, List.class)) {
+            return ListHelper.getList(resourceResolver, annotation.value(), getTypeArgument(type, 0));
+
+        } else if (InjectorUtils.isValidMapKeyType(type, String.class)) {
+            return getTypeArgument(type, 1).equals(String.class)
+                ? ListHelper.getMap(resourceResolver, annotation.value())
+                : ListHelper.getMap(resourceResolver, annotation.value(), JcrConstants.JCR_TITLE, getTypeArgument(type, 1));
+
+        } else if (!(type instanceof ParameterizedType) && ((Class<?>) type).isArray()) {
+            List<?> list = ListHelper.getList(resourceResolver, annotation.value(), ((Class<?>) type).getComponentType());
+            return toArray(list, ((Class<?>) type).getComponentType());
+        }
+
+        LOG.debug(InjectorConstants.EXCEPTION_UNSUPPORTED_TYPE, type);
+        return null;
+    }
+
+    /**
+     * Retrieves type arguments from parameterized collection type
+     * @param type  Type of receiving Java class member
+     * @param index The index of the element in the array of type objects
+     * @return Class of type object
+     */
+    private Class<?> getTypeArgument(Type type, int index) {
+        return (Class<?>) ((ParameterizedType) type).getActualTypeArguments()[index];
+    }
+
+    /**
+     * Converts the List of unknown to an array and casts it to the specified type
```

Comment:
"List of unknown" sounds... so poetic))
"A generic {@code List}"

Reply:
Line 162 still contains the unknown :)